### PR TITLE
Configure vagrant to sync only the files and folders that are needed for deploying the web application

### LINKED ISSUE #136: Configure vagrant to sync only the files and folders for the web application deployment

### PR DESCRIPTION
**Pull request by ipolonsk**
FIx #136 
Configure vagrant to sync only the files and folders that are needed for deploying and bringing up the app application on the AWS instance.
Files and folders that are synced:
requirements.txt, app.py, baboon/

Files and folders that are not synced: 
.env, .env_template, .git/, .github/, .gitignore, CONTRIBUTORS.md, LICENSE.md, README.md, pytest.ini, tests/, venv/, .idea/, Vagrantfile, bootstrap-infra.py, bootstrap.sh

I tested the deployment with this liner(dev and stage), and the app was successfully up and accessible.

### CODE REVIEW

**Comment by IlanZuckerman** [comment]
LGTM +1

File: Vagrantfile [hunk 1]
```diff
@@ -11,7 +11,11 @@ end
 
 Vagrant.configure("2") do |config|
   config.vagrant.plugins= ["vagrant-env"]
-  config.vm.synced_folder ".", "/vagrant", type: "rsync", rsync__exclude: [ ".env", ".env_template", ".git/", ".github/", ".gitignore", "CONTRIBUTORS.md", "LICENSE.md", "README.md", "pytest.ini", "tests/", "venv/", ".idea/", "Vagrantfile", "bootstrap-infra.py", "bootstrap.sh" ]
+  config.vm.synced_folder ".", "/vagrant", type: "rsync", disabled: true
+  config.vm.synced_folder "baboon/", "/vagrant/baboon/", type: "rsync"
+  config.vm.provision "file", source: "requirements.txt", destination: "/tmp/requirements.txt"
+  config.vm.provision "file", source: "app.py", destination: "/tmp/app.py"
+  config.vm.provision "shell", inline: "sudo cp /tmp/{requirements.txt,app.py}  /vagrant/"
```
Comment (lioramilbaum) [inline Vagrantfile:18]: This approach was implemented in an inconsistent way. It will be very difficult to maintain it.

Comment (ipolonsk) [reply Vagrantfile:18]: The third option is to create a folder that will hold all the source code that needed for the app, and then only run the syned_folder on that folder, but it requires a bigger change to the folders structure and update the related code to use the new paths.

Comment (lioramilbaum) [reply Vagrantfile:18]: Than go to the first approach and exclude only big folders like: .git. That should suffice.

**Pull request merged**